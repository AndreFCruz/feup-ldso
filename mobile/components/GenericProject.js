import React from 'react';
import PropTypes from 'prop-types';
import { material } from 'react-native-typography';
import { ImageBackground, View, Text, StyleSheet, Image, TouchableOpacity, Dimensions } from 'react-native';
import ActivityIndicatorView from './ActivityIndicatorView';
import { Icon, Badge } from 'native-base';
import { connect } from 'react-redux';
import { facultyStyles } from '../constants/SpecificStyles';

class GenericProject extends React.Component {

    componentDidMount() {
        const { language, faculty, id } = this.props;

        this.props.getOne(language, faculty, id);
    }

    componentDidUpdate(prevProps) {
        const { language, faculty, id } = this.props;

        if (prevProps.language != language || prevProps.id != id) 
            this.props.getOne(language, faculty, id);
    }

    render() {
        const { project, loading, faculty, language, navActionDown, navActionUp } = this.props;

        if (loading || !project)
            return ( <ActivityIndicatorView></ActivityIndicatorView> );

        return (
            <View>
                <ImageBackground source={require('../assets/images/background.png')} style={styles.fullScreen}>
                    <View style={styles.container}>
                        <View style={styles.topLeftChild}>
                            <Image source={{uri: project.images[0]}} style={[styles.topLeftImage, styles.icon, facultyStyles[faculty].icon]} />
                        </View>
                        <View style={styles.topRightChild}>
                            <View style={styles.topRightHeader}>
                                <View style={styles.statusButtonContainer}>
                                    { project.hasOwnProperty('active') &&
                                    <Badge style={[styles.statusButton, project.active ? styles.activeButton : styles.inactiveButton]}> 
                                        <Text style={styles.statusButtonText}>
                                            { language == 'en' ? project.active ? 'Active' : 'Inactive' : project.active ? 'Ativo' : 'Inativo '}
                                        </Text> 
                                    </Badge>
                                    }
                                </View>
                                <Text style={styles.heading}>{project.title}</Text>
                                <Text style={styles.body}>{project.short_description}</Text>
                            </View>
                            <View style={styles.topRightChildImage}>
                                <Image source={{uri: project.images[1]}} style={[styles.topRightImage, styles.icon, facultyStyles[faculty].icon]} />
                            </View>
                        </View>
                    </View>
                    <View style={styles.topNavigation}>
                        <TouchableOpacity onPress={navActionDown}>
                            <Icon type="FontAwesome" name="chevron-down"/>
                        </TouchableOpacity>
                    </View>
                </ImageBackground>
                <View style={[styles.fullScreen, styles.darkBackground]}>
                    <View style={styles.bottomNavigation}>
                        <TouchableOpacity onPress={navActionUp}>
                            <Icon type="FontAwesome" name="chevron-up" style={styles.whiteArrow} />
                        </TouchableOpacity>
                    </View>
                    <View style={styles.container} >
                        <View style={styles.bottomLeftChild}>
                            <Text style={[styles.mainText, styles.body, styles.whiteFont]}>{project.content}</Text>
                        </View>
                        <View style={styles.bottomRightChild}>
                            <Image source={{uri: project.images[1]}} style={[styles.bottomRightImage, styles.icon, facultyStyles[faculty].icon]} />
                        </View>
                    </View>
                </View>
            </View>
        );
    }
}

GenericProject.propTypes = {
    id: PropTypes.string,
    faculty: PropTypes.string,
    loading: PropTypes.bool,
    project: PropTypes.object,
    language: PropTypes.string,
    getOne: PropTypes.func,
    navActionUp: PropTypes.func,
    navActionDown: PropTypes.func
};

const styles = StyleSheet.create({
    // Global
    fullScreen: {
        height: Dimensions.get('window').height,
    },
    container: {
        flex: 5,
        flexDirection: 'row',
    },
    icon: {
        borderRadius: 25,
        borderWidth: 5
    },

    // Top View
    statusButtonContainer: {
        alignSelf: 'flex-end',
    },
    statusButton: {
        marginTop: 10,
        marginRight: 15,
    },
    statusButtonText: {
        margin: 3,
        color: 'white'
    },
    activeButton: {
        backgroundColor: 'green'
    },
    inactiveButton: {
        backgroundColor: 'red'
    },
    topNavigation: {
        flex: 1,
        alignItems: 'center' 
    },
    topLeftChild: {
        margin: 10,
        flex: 1,
    },
    topRightChild: {
        margin: 10,
        flex: 1,
        alignSelf: 'center',
        justifyContent: 'flex-end'
    },
    topLeftImage: {
        flex: 1,
        margin: 10,
        width: undefined,
        height: undefined
    },
    topRightHeader: {
        flex: 2,
    },
    topRightChildImage: {
        flex: 3,
    },
    topRightImage: {
        flex: 1,
        margin: 10,
        marginTop: 25,
        width: undefined,
        height: undefined
    },

    //Bottom View
    whiteArrow: {
        color: 'white'
    },
    darkBackground: {
        backgroundColor: '#1c1c1c'
    },
    bottomNavigation: {
        marginTop: 10,
        flex: 1,
        alignItems: 'center'
    },
    bottomLeftChild: {
        marginVertical: 10,
        marginHorizontal: 15,
        flex: 1,
        alignItems: 'center'
    },
    bottomRightChild: {
        margin: 10, 
        flex: 1,
        justifyContent: 'flex-start'
    },
    bottomRightImage: {
        flex: 0.5,
        margin: 10,
        width: undefined,
        height: undefined
    },

    // Typography
    heading: {
        ...material.display1Object,
        fontFamily: 'Quicksand_regular',
        marginTop: 'auto',
        marginBottom: 20,
    },
    body: {
        ...material.subheadingObject,
        fontFamily: 'OpenSans_regular',
        fontSize: 20,
        lineHeight: 30
    },
    whiteFont: {
        color: 'white'
    },
    mainText: {
        maxWidth: 500
    }
});

const mapStateToProps = ({ faculty, language }) => ({
    faculty: faculty.name,
    loading: faculty.loading,
    language: language.selection
});

export default connect(mapStateToProps, null)(GenericProject);