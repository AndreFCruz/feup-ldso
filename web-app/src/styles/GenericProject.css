.prospectsParent,
.holderParent {
    width: 100%;
    height: 68vh;
    overflow: hidden;
}

.prospectsParent {
    margin-left: 0.5em;
}

.holderParent {
    margin-left: 2em;
}

.projectHolder {
    width: 100%;
    height: 100%;
    overflow-y: scroll;
    padding-right: 17px;
    box-sizing: content-box;
}

.versions * {
    font-weight: 600;
}

.center-text {
    text-align: center;
}

.row-style {
    margin-bottom: 2em;
}

.projectHolder textarea {
    height: 150px !important;
    resize: none;
}

.projectHolder textarea.contentEditor {
    height: 300px !important;
}

.dateInput {
    margin: 0 auto;
    width: auto !important;
}

.mainActionBtn {
    width: 180px !important;
    font-size: 20px !important;
    padding: 0.5em 2em !important;
    transition: background-color 1s, width 200ms !important;
}

.mainActionBtn:hover {
    background-color: green !important;
}

.editProspectsBtn {
    width: 250px !important;
}

/* Loading effect */
@-webkit-keyframes ld {
    0% {
        transform: rotate(0deg) scale(1);
    }

    50% {
        transform: rotate(180deg) scale(1.1);
    }

    100% {
        transform: rotate(360deg) scale(1);
    }
}

@-moz-keyframes ld {
    0% {
        transform: rotate(0deg) scale(1);
    }

    50% {
        transform: rotate(180deg) scale(1.1);
    }

    100% {
        transform: rotate(360deg) scale(1);
    }
}

@-o-keyframes ld {
    0% {
        transform: rotate(0deg) scale(1);
    }

    50% {
        transform: rotate(180deg) scale(1.1);
    }

    100% {
        transform: rotate(360deg) scale(1);
    }
}

@keyframes ld {
    0% {
        transform: rotate(0deg) scale(1);
    }

    50% {
        transform: rotate(180deg) scale(1.1);
    }

    100% {
        transform: rotate(360deg) scale(1);
    }
}

.m-progress {
    position: relative;
    opacity: .8;
    color: transparent !important;
    text-shadow: none !important;
}

.m-progress:hover,
.m-progress:active,
.m-progress:focus {
    cursor: default;
    color: transparent;
    outline: none !important;
    box-shadow: none;
}

.m-progress:before {
    content: '';

    display: inline-block;

    position: absolute;
    background: transparent;
    border: 1px solid #fff;
    border-top-color: transparent;
    border-bottom-color: transparent;
    border-radius: 50%;

    box-sizing: border-box;

    top: 50%;
    left: 50%;
    margin-top: -12px;
    margin-left: -12px;

    width: 24px;
    height: 24px;

    -webkit-animation: ld 1s ease-in-out infinite;
    -moz-animation: ld 1s ease-in-out infinite;
    -o-animation: ld 1s ease-in-out infinite;
    animation: ld 1s ease-in-out infinite;
}
