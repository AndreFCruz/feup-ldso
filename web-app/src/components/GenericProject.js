import React, { Component } from 'react';
import { Input, Col, Row, Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import withRouter from 'react-router-dom/withRouter';
import PropTypes from 'prop-types';

import '../styles/GenericProject.css';

class GenericProject extends Component {
    constructor(props) {
        super(props);

        this.state = {
            titleEN: '',
            titlePT: '',
            contentEN: '',
            contentPT: '',
            descriptionEN: '',
            descriptionPT: '',
            images: '',
            startDate: '',
            endDate: '',
            modal: false
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.toggle = this.toggle.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.onDelete = this.onDelete.bind(this);
    }

    componentDidUpdate(prevProps) {
        const { projEN, projPT, loadingAction } = this.props;

        if (projEN != null && projPT != null && (prevProps.projEN != projEN || prevProps.projPT != projPT)) {
            this.setState({
                titleEN: projEN.title,
                titlePT: projPT.title,
                contentEN: projEN.content,
                contentPT: projPT.content,
                descriptionEN: projEN.short_description,
                descriptionPT: projPT.short_description,
                images: projEN.images != undefined? projEN.images.reduce((acc, val) => `${acc}\n${val}`) : '',
                startDate: projEN.start_date,
                endDate: projEN.end_date
            });
        }

        if (prevProps.loadingAction !== loadingAction && this.state.modal)
            this.toggle();
    }

    toggle() {
        this.setState({
            modal: !this.state.modal
        });
    }

    onSubmit() {
        const { add, faculty, projEN, projPT, mainAction, history } = this.props;
        let imagesArray = this.state.images.split('\n');
        imagesArray = imagesArray.map(el => el.trim());

        if (add) {
            mainAction(faculty, {
                ...this.state,
                images: imagesArray
            });
            history.push(`/faculties/${faculty}/projects`);
        } else {
            mainAction(faculty, 'en', projEN.id, {
                title: this.state.titleEN,
                content: this.state.contentEN,
                short_description: this.state.descriptionEN,
                images: imagesArray,
                start_date: this.state.startDate,
                end_date: this.state.end_date
            });
            mainAction(faculty, 'pt', projPT.id, {
                title: this.state.titlePT,
                content: this.state.contentPT,
                short_description: this.state.descriptionPT,
                images: imagesArray,
                start_date: this.state.startDate,
                end_date: this.state.end_date
            });
        }
    }

    onDelete() {
        const { delAction, projEN, projPT } = this.props;

        delAction(projEN.id);
        delAction(projPT.id);
    }

    handleInputChange(event) {
        this.setState({
            [event.target.name]: event.target.value
        });
    }

    render() {
        const { loading, loadingAction, add, projEN, projPT, delAction } = this.props;

        const deleteHtml = delAction != null ? <Button outline color="danger" onClick={this.toggle}>Delete</Button> : '';

        if (loading) {
            return <div>Loading...</div>;
        } else if (projEN == null || projPT == null) {
            return <div></div>;
        } else {
            return (
                <div className="holderParent">
                    <div className="projectHolder">
                        <Row className="row-style">
                            <Col className="center-text">
                                <h2>{add ? 'New' : 'Edit'} Social Impact Project{delAction != null? ' | ' : ''}{deleteHtml}</h2>
                            </Col>
                        </Row>
                        <Row className="versions row-style">
                            <Col xs="6" className="center-text">
                                <h3>English Version</h3>
                            </Col>
                            <Col xs="6" className="center-text">
                                <h3>Versão Portuguesa</h3>
                            </Col>
                        </Row>
                        <Row>
                            <Col xs="6" className="center-text">
                                <h4>Title</h4>
                            </Col>
                            <Col xs="6" className="center-text">
                                <h4>Título</h4>
                            </Col>
                        </Row>
                        <Row className="row-style">
                            <Col xs="6" className="center-text">
                                <Input name="titleEN"
                                    value={this.state.titleEN}
                                    onChange={this.handleInputChange} />
                            </Col>
                            <Col xs="6" className="center-text">
                                <Input name="titlePT"
                                    value={this.state.titlePT}
                                    onChange={this.handleInputChange} />
                            </Col>
                        </Row>
                        <Row>
                            <Col xs="6" className="center-text">
                                <h4>Short Description</h4>
                            </Col>
                            <Col xs="6" className="center-text">
                                <h4>Pequena Descrição</h4>
                            </Col>
                        </Row>
                        <Row className="row-style">
                            <Col xs="6" className="center-text">
                                <Input name="descriptionEN"
                                    value={this.state.descriptionEN}
                                    onChange={this.handleInputChange} />
                            </Col>
                            <Col xs="6" className="center-text">
                                <Input name="descriptionPT"
                                    value={this.state.descriptionPT}
                                    onChange={this.handleInputChange} />
                            </Col>
                        </Row>
                        <Row>
                            <Col xs="6" className="center-text">
                                <h4>Content</h4>
                            </Col>
                            <Col xs="6" className="center-text">
                                <h4>Conteúdo</h4>
                            </Col>
                        </Row>
                        <Row className="row-style">
                            <Col xs="6" className="center-text">
                                <Input type="textarea" name="contentEN"
                                    value={this.state.contentEN}
                                    onChange={this.handleInputChange} />
                            </Col>
                            <Col xs="6" className="center-text">
                                <Input type="textarea" name="contentPT"
                                    value={this.state.contentPT}
                                    onChange={this.handleInputChange} />
                            </Col>
                        </Row>
                        <Row>
                            <Col className="center-text">
                                <h4>Images / Imagens</h4>
                            </Col>
                        </Row>
                        <Row className="row-style">
                            <Col className="center-text">
                                <Input type="textarea" name="images"
                                    value={this.state.images}
                                    onChange={this.handleInputChange} />
                            </Col>
                        </Row>
                        <Row>
                            <Col xs="6" className="center-text">
                                <h4>Begin Date / Data de Início</h4>
                            </Col>
                            <Col xs="6" className="center-text">
                                <h4>End date / Data de Fim</h4>
                            </Col>
                        </Row>
                        <Row className="row-style">
                            <Col xs="6" className="center-text">
                                <Input type="date" className="dateInput"
                                    name="startDate"
                                    value={this.state.startDate === undefined? '' : this.state.startDate}
                                    onChange={this.handleInputChange} />
                            </Col>
                            <Col xs="6" className="center-text">
                                <Input type="date" className="dateInput"
                                    name="endDate"
                                    value={this.state.endDate === undefined? '' : this.state.endDate}
                                    onChange={this.handleInputChange} />
                            </Col>
                        </Row>
                        <Row>
                            <Col className="center-text">
                                <Button className={`${loadingAction ? 'm-progress' : ''} mainActionBtn`}
                                    onClick={this.onSubmit}>
                                    {add ? 'Create' : 'Save'} Project
                                </Button>
                            </Col>
                        </Row>
                    </div>
                    <Modal isOpen={this.state.modal} toggle={this.toggle}>
                        <ModalHeader toggle={this.toggle}>Delete Project</ModalHeader>
                        <ModalBody>
                            This action is irreversible. Are you sure you want to proceed?
                        </ModalBody>
                        <ModalFooter>
                            <Button outline color="danger" className={loadingAction ? 'm-progress' : ''} onClick={this.onDelete}>Delete</Button>{' '}
                            <Button color="secondary" onClick={this.toggle}>Cancel</Button>
                        </ModalFooter>
                    </Modal>
                </div>
            );
        }
    }
}

GenericProject.propTypes = {
    history: PropTypes.object,
    add: PropTypes.bool,
    loading: PropTypes.bool,
    loadingAction: PropTypes.bool,
    faculty: PropTypes.string,
    projEN: PropTypes.object,
    projPT: PropTypes.object,
    mainAction: PropTypes.func,
    delAction: PropTypes.func
};

export default withRouter(GenericProject);
