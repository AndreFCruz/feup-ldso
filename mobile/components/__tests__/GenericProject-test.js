import 'react-native';
import React from 'react';
import GenericProject from '../GenericProject';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(loading = false) {

    const initialState = { 
        faculty: {
            loading: loading,
            name: 'feup'
        },
        language: {
            selection: 'en'
        }
    };

    const props = {
        id: '123',
        getOne: jest.fn(),
        project: {
            'title': '[EN] Social project at FAUP',
            'short_description': 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Etiam neque. Aliquam erat volutpat. Suspendisse sagittis ultrices augue. Pellentesque ipsum.',
            'content': '[EN] Aliquam sit amet varius lacus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tortor justo, ornare a vulputate in, mattis id mi. Donec varius, dui venenatis pharetra ultricies, ligula nulla ornare ex, quis mattis libero quam a tellus. Quisque quis est id nulla ornare semper at a nisi. Integer tincidunt egestas ornare. Nullam congue tellus arcu, eu ultricies mi pellentesque eget. Pellentesque massa sapien, suscipit sit amet lacus vel, faucibus faucibus elit. Cras in metus viverra, luctus magna ac, tristique leo. Aenean ultrices dui eget velit cursus, vitae suscipit purus varius. Aliquam erat volutpat. In porta metus et velit aliquet porta.',
            'start_date': '2017-01-01',
            'images': [
                'https://picsum.photos/200/300/?random',
                'https://picsum.photos/112/200/?random'
            ],  
            'facultyId': 1
        }
    };

    const wrapper = shallow(<GenericProject {...props} />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return wrapper;
}

it('renders loading screen', () => {
    const wrapper = setup(true);

    expect(wrapper.dive()).toMatchSnapshot();
});
    
it('renders screen with geric project content', () => {
    const wrapper = setup();

    expect(wrapper.dive()).toMatchSnapshot();
});

