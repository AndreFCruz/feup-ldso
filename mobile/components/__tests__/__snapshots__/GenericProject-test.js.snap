// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading screen 1`] = `<ActivityIndicatorView />`;

exports[`renders screen with geric project content 1`] = `
<Component>
  <ImageBackground
    source={1}
    style={
      Object {
        "height": 1334,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "margin": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://picsum.photos/200/300/?random",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": undefined,
                "margin": 10,
                "width": undefined,
              },
              Object {
                "borderRadius": 25,
                "borderWidth": 5,
              },
              Object {
                "borderColor": "#9f2d20",
              },
            ]
          }
        />
      </Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <Component
            style={
              Object {
                "alignSelf": "flex-end",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0,0,0,0.54)",
                "fontFamily": "Quicksand_regular",
                "fontSize": 34,
                "fontWeight": "400",
                "letterSpacing": 0.381836,
                "lineHeight": 40,
                "marginBottom": 20,
                "marginTop": "auto",
              }
            }
          >
            [EN] Social project at FAUP
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0,0,0,0.87)",
                "fontFamily": "OpenSans_regular",
                "fontSize": 20,
                "fontWeight": "400",
                "letterSpacing": -0.32,
                "lineHeight": 30,
              }
            }
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Etiam neque. Aliquam erat volutpat. Suspendisse sagittis ultrices augue. Pellentesque ipsum.
          </Text>
        </Component>
        <Component
          style={
            Object {
              "flex": 3,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://picsum.photos/112/200/?random",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "margin": 10,
                  "marginTop": 25,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 25,
                  "borderWidth": 5,
                },
                Object {
                  "borderColor": "#9f2d20",
                },
              ]
            }
          />
        </Component>
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Styled(Icon)
          name="chevron-down"
          type="FontAwesome"
        />
      </TouchableOpacity>
    </Component>
  </ImageBackground>
  <Component
    style={
      Array [
        Object {
          "height": 1334,
        },
        Object {
          "backgroundColor": "#1c1c1c",
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 10,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Styled(Icon)
          name="chevron-up"
          style={
            Object {
              "color": "white",
            }
          }
          type="FontAwesome"
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginHorizontal": 15,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "maxWidth": 500,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0,0,0,0.87)",
                "fontFamily": "OpenSans_regular",
                "fontSize": 20,
                "fontWeight": "400",
                "letterSpacing": -0.32,
                "lineHeight": 30,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          [EN] Aliquam sit amet varius lacus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tortor justo, ornare a vulputate in, mattis id mi. Donec varius, dui venenatis pharetra ultricies, ligula nulla ornare ex, quis mattis libero quam a tellus. Quisque quis est id nulla ornare semper at a nisi. Integer tincidunt egestas ornare. Nullam congue tellus arcu, eu ultricies mi pellentesque eget. Pellentesque massa sapien, suscipit sit amet lacus vel, faucibus faucibus elit. Cras in metus viverra, luctus magna ac, tristique leo. Aenean ultrices dui eget velit cursus, vitae suscipit purus varius. Aliquam erat volutpat. In porta metus et velit aliquet porta.
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "margin": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://picsum.photos/112/200/?random",
            }
          }
          style={
            Array [
              Object {
                "flex": 0.5,
                "height": undefined,
                "margin": 10,
                "width": undefined,
              },
              Object {
                "borderRadius": 25,
                "borderWidth": 5,
              },
              Object {
                "borderColor": "#9f2d20",
              },
            ]
          }
        />
      </Component>
    </Component>
  </Component>
</Component>
`;
